### PR TITLE
[docs] Revise Okta SAML docs

### PR DESCRIPTION
**Pull request by alexyarosh**
- Add more detailed and clearly outlined steps 
- Fix the group config docs as outlined in #40517
- Add Okta config screenshots because they were requested in many tickets

### CODE REVIEW

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:29]:
```suggestion
Before configuring SAML authentication in Metabase, you'll need to create a new SAML app integration in Okta.
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:31]:
```suggestion
### Create an app integration in Okta
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:37]:
```suggestion
To configure Okta app integration with Metabase, you'll need to use the information found in Metabase in the **Admin panel** > **Authentication** > **SAML** section.
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:56]:
```suggestion
In the **Attribute statements (optional)** section of the Okta application SAML setting, create the following attribute statements:

- email address
- first name (given name)
- last name (surname)

Even though Okta says these are optional, Metabase requires them. Okta will pass these attributes to Metabase during authentication to automatically log people in to Metabase.
```

The original phrasing (which repeats the attributes) could be confusing given the conflict between "requires" and "optional". The rephrasing omits the repetition of attributes and explains the requirement.

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:64]: Should we mention case sensitivity? And is non-default the same as "custom"? What about additional?

```suggestion
The names of attribute statement in Okta should match the attribute names in Metabase (names are case sensitive). If you want to use custom attribute names in your Okta app configuration, you'll also need to change the names for the attribute fields in Metabase: **Admin panel** > **Authentication** > **SAML**.
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:66]:
```suggestion
> **Make sure that people cannot edit their email address attribute**. To log people in to your Metabase (or to create a Metabase account on first login), your IdP will pass the email address attribute to Metabase. If a person can change the email address attribute, they'll potentially be able to access Metabase accounts other than their own.
```

Also: how should an admin make sure that people can't edit their email address attribute? Is there an explicit action the admin needs to take to ensure that the email attribute is uneditable by account holders?

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:114]:
```suggestion
Once you set up your SAML app in Okta, you'll need to configure SAML in Metabase. You'll need some information from Okta:

1. In Okta, go to the page for your Metabase app integration.
2. Go to the **Sign On** tab.
3. Click on **View SAML setup instructions**.
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:128]:
```suggestion
You can configure Metabase to automatically assign people to Metabase groups when they log in. You'll need to create a SAML attribute statement that will pass the groups information to Metabase, and then configure Metabase to read this attribute and map its contents to Metabase groups.
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:130]:
```suggestion
You can use either:

-  [A custom user profile attribute](#use-a-user-profile-attribute-to-assign-groups) that contains user's Metabase groups.
-  [Okta User Groups](#map-okta-user-groups-to-metabase-groups).
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:144]:
```suggestion
   Metabase groups don't have to correspond to Okta User Groups. If you'd like to use Okta User Groups to set up Metabase Groups, see [Map Okta User Groups to Metabase groups](#map-okta-user-groups-to-metabase-groups).
```

Yes, "leverage" avoids the "use ... User" situation, but it's still pretty corporatese.

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:146]:
```suggestion
   > Your Okta account has to have `SAML_SUPPORT_ARRAY_ATTRIBUTES` enabled, as Metabase expects Okta to pass attributes as an array. If your Okta account is old, you might need to reach out to Okta support to enable `SAML_SUPPORT_ARRAY_ATTRIBUTES`.
```

This phrasing frontloads the requirement, followed by the explanation.

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:156]:
```suggestion
- Turn on **Synchronize Group Memberships**.
- For each of the groups you added to Okta users, set up a new mapping to a Metabase group.
- In **Group attribute name**, enter `MetabaseGroupName` (the name of the SAML attribute statement).
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:163]:
```suggestion
2. In Okta's **SAML Settings** for the Metabase app integration, add a new attribute statement `MetabaseGroupName`, set the type to "Basic", and the value to:
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:177]:
```suggestion
   > Your Okta account has to have `SAML_SUPPORT_ARRAY_ATTRIBUTES` enabled, as Metabase expects Okta to pass attributes as an array. If your Okta account is old, you might need to reach out to Okta support to enable `SAML_SUPPORT_ARRAY_ATTRIBUTES`.
```

Same as above.

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:185]:
```suggestion
- Turn on **Synchronize Group Memberships**.
- For each of the groups you added to Okta users, set up a new mapping to a Metabase group.
- In **Group attribute name**, enter `MetabaseGroupName` (the name of the SAML attribute statement).
```

File: docs/people-and-groups/saml-okta.md
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-okta.md:171]: Can we make this a code block? I'm not sure why it's an image.

**Review by jeff-bruemmer — APPROVED** [review]
Approving to unblock, but please address comments and suggestions before merging.

This is a big improvement on our Okta docs. Nice work, @alexyarosh.

File: docs/people-and-groups/saml-okta.md
Comment (alexyarosh) [reply docs/people-and-groups/saml-okta.md:66]: @jeff-bruemmer To be honest, I am not sure, I kept this as-is from the previous version of the docs. We say this in all SAML docs but don't  explain how to make this happen. I think it's a property of the user profile attribute in the IdP

File: docs/people-and-groups/saml-okta.md
Comment (alexyarosh) [reply docs/people-and-groups/saml-okta.md:64]: "custom attribute names" had the ambiguity whether it's `(custom attribute) names` or `custom (attribute names)` because there's a concept of a custom attribute (and it's even something that we reference later in the article). 

But in our case it's about using `(custom name)` for an attribute statement (and those attribute statements check regular, non-custom attributes).

There were customers who used different (from MB-provided default) names in Okta and had problems. They just needed to copy paste the same names back into MB

**Review by Tony-metabase — APPROVED** [review]
Love it!

File: docs/people-and-groups/saml-okta.md
Comment (alexyarosh) [inline docs/people-and-groups/saml-okta.md:182]:
```suggestion
   Next, you'll need to tell Metabase how to map Okta groups to Metabase groups.
```

**Pull request merged**
